model/user.go

```diff
@@ -391,7 +391,7 @@ func (u *User) SetDefaultNotifications() {
 	u.NotifyProps[PUSH_NOTIFY_PROP] = USER_NOTIFY_MENTION
 	u.NotifyProps[DESKTOP_NOTIFY_PROP] = USER_NOTIFY_MENTION
 	u.NotifyProps[DESKTOP_SOUND_NOTIFY_PROP] = "true"
-	u.NotifyProps[MENTION_KEYS_NOTIFY_PROP] = u.Username + ",@" + u.Username
+	u.NotifyProps[MENTION_KEYS_NOTIFY_PROP] = ""
```

Comment:
Looks like there is another occurrence of this default prop value, mind confirming if that needs to be updated? I believe that gets invoked when updating the username.

It's on line: https://github.com/mattermost/mattermost-server/pull/13851/files#diff-9a8b5fc2b9714b5cb3702424abb321b0R409

Reply:
oh - interesting! Thanks so much for the double check.  I'll try update my username so I can be sure to test the code before and after I change it.  Stay tuned!

Should we consider adding something to [user_test.go](https://github.com/mattermost/mattermost-server/blob/b0dd349b9ef95e07d055c349731ef3444b82b811/model/user_test.go) so that if we change this again in the future CI will catch it not being updated?

Reply:
Thanks @Ths2-9Y-LqJt6! 

And yes, once you confirm the behaviour that updating the username resets `mention_keys` to the unintended default value, then yes, please add a test to cover that case. 👍

Reply:
@ali-farooq0 - nice catch with the deceleration in `UpdateMentionKeysFromUsername()`! I've gone ahead and made a change (adf11574403620ebe1287410880488f7c83a88a5) and confirmed before and after that it should now work as expected.  

I'll dig into writing a test to catch it next.  V. exciting...writing actual net new Go code!! I'll report back when I have a test for it (or if I need help ; )

Reply:
@ali-farooq0 - oh, I some how missed that there were tests already! Indeed, they were failing with the new code change. Everything should be good with 57ae393ac0247a5b520e3ead6a8e211691ac6863, but lemme know if not!

Reply:
Excellent! 👍